ground/src/main/java/com/google/android/ground/ui/common/ViewModelFactory.kt

```diff
@@ -37,8 +36,22 @@ constructor(
   /** Instantiates a new instance of the specified view model, injecting required dependencies. */
   override fun <T : ViewModel> create(modelClass: Class<T>): T {
     val creator =
-      creators[modelClass] ?: throw IllegalArgumentException("Unknown model class $modelClass")
-    return creator.get() as T
+      creators[modelClass]
+        ?: creators.entries.firstOrNull { modelClass.isAssignableFrom(it.key) }?.value
+        ?: error("Unknown model class $modelClass")
+    try {
+      val viewModel = creator.get()
+      if (modelClass.isInstance(viewModel)) {
+        @Suppress("UNCHECKED_CAST")
+        return viewModel as T
+      } else {
+        error(
+          "Expected instance of ${modelClass.simpleName}, but got ${viewModel::class.simpleName}"
+        )
+      }
+    } catch (e: Exception) {
+      throw ViewModelCreationException("Failed to create ViewModel for class $modelClass", e)
+    }
```

Comment:
Why are we catching and re-throwing the errors?

Reply:
The initial idea of having 3 throws:
1. if the modelClass is unknown
2. if the modelClass is not a viewModel class type then throw in else part
3. If we get any error while `val viewModel = creator.get()` getting then our custom throw `ViewModelCreationException`.

Reply:
I would argue that it is better to throw directly and now catch and rethrow as it doesn't provide any extra benefits. We won't be using `ViewModelCreationException` for any special handling.

Reply:
fixed by removing the extra throw